pandas/core/ops.py

```diff
@@ -703,6 +702,39 @@ def wrapper(left, right, name=name, na_op=na_op):
     return wrapper
 
 
+def dispatch_to_index_op(op, left, right, index_class):
+    """
+    Wrap Series left in the given index_class to delegate the operation op
+    to the index implementation.  DatetimeIndex and TimedeltaIndex perform
+    type checking, timezone handling, overflow checks, etc.
+
+    Parameters
+    ----------
+    op : binary operator (operator.add, operator.sub, ...)
+    left : Series
+    right : object
+    index_class : DatetimeIndex or TimedeltaIndex
+
+    Returns
+    -------
+    result : object, usually DatetimeIndex, TimedeltaIndex, or Series
+    """
+    left_idx = index_class(left)
+
+    # avoid accidentally allowing integer add/sub.  For datetime64[tz] dtypes,
+    # left_idx may inherit a freq from a cached DatetimeIndex.
+    # See discussion in GH#19147.
+    left_idx.freq = None
```

Comment:
actually I don't like this mutation here. why is it necessary? (e.g. if one of the end-points is None, e.g. an integer series this will still raise, yes?)

Reply:
Because if `left` is tzaware then `pd.DatetimeIndex(left, freq=None)` may come back with non-None freq.

```
dti = pd.date_range('2016-01-01', periods=2, tz='US/Pacific')
ser = pd.Series(dti)

>>> pd.DatetimeIndex(ser, freq=None).freq
<Day>
```

Reply:
and why is this a problem?

Reply:
Because in this scenario, without the `left_idx.freq = None` line, `ser + 1` fails to raise.

Reply:
Just parametrized `test_dt64_series_add_intlike` to make sure the tzaware case is covered.

Reply:
so you didn't answer my question here. I don't want mutation.

Reply:
I really, really did.  Without setting the frequency to none, integer addition fails to raise.

Reply:
then we need a better way to detect this, we cannot mutate things to just see if they raise